src/main/java/org/jabref/gui/push/PushToApplicationAction.java

```diff
@@ -25,25 +25,37 @@
  */
 public class PushToApplicationAction extends SimpleCommand {
 
-    private final PushToApplication operation;
     private final StateManager stateManager;
     private final DialogService dialogService;
 
+    private PushToApplication application;
+    private Action action;
+
     public PushToApplicationAction(StateManager stateManager, PushToApplicationsManager pushToApplicationsManager, DialogService dialogService) {
-        this.operation = pushToApplicationsManager.getLastUsedApplication(Globals.prefs);
+        this.application = pushToApplicationsManager.getActiveApplication(Globals.prefs);
         this.stateManager = stateManager;
         this.dialogService = dialogService;
 
         this.executable.bind(needsDatabase(stateManager).and(needsEntriesSelected(stateManager)));
         this.statusMessage.bind(BindingsHelper.ifThenElse(this.executable, "", Localization.lang("This operation requires one or more entries to be selected.")));
+
+        updateAction();
+    }
+
+    public void updateApplication (PushToApplication application) {
+        this.application = application;
+        updateAction();
     }
 
     public Action getActionInformation() {
-        return new Action() {
+        return action;
```

Comment:
What was the reason to change `getActionInformation` (i.e. instead of returning a new instance of the action, keep one and update it)?

Reply:
It keeps the toolBarButton and the menuItem in sync with the same action.

Reply:
You were absolutely right. It was an artifact of an earlier version, when I needed it as a persistent action. I reverted it and it still works fine.